allow empty environment configuration

l8nite opened the pull request:
I had an empty block in my config/application.yml and figaro crashed:
```
/home/sguth/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/figaro-1.1.0/lib/figaro/application.rb:34:in `merge': no implicit conversion of nil into Hash (TypeError)
        from /home/sguth/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/figaro-1.1.0/lib/figaro/application.rb:34:in `configuration'
```

This is a small patch to let it continue.

![Cat](http://i.imgur.com/LWMA0TL.gif)

laserlemon commented:
Thank you! :clap: A few suggestions:

1. Let's move the hash default into the `environment_configuration` method itself.
2. Are you comfortable adding a test to exercise this behavior so we don't introduce a regression later on?
3. You seemed to have skipped step 4 in the [contribution guidelines].

l8nite commented:
@laserlemon thanks - I moved the defaulting behavior to `environment_configuration` and introduced a spec, sorry for not including that in the original change. I also added a gif. :yellow_heart:

laserlemon commented:
Thank you! I made some very slight style adjustments and merged into master!

laserlemon commented:
Version 1.1.1 shipped!

laserlemon commented:
Also, great gif!
